### ISSUE TITLE
Intake Usage

### ISSUE THREAD

**Issue by MadgePie** [informatics-lab/aml-jupyterhub#25]
Looking into how intake may interact with Azure.

**Comment by tam203** [comment]
Consider:
* Data sets / data stores
* @stevehadd  might have some insight
* keen on intake over Az specific technologies but might be good to stack on on other?

**Comment by tam203** [comment]
@DPeterK - @MadgePie  is starting to look into intaking all the data. Pls coordinate with the datasets you are looking to provide/expose.

**Comment by DPeterK** [comment]
My feeling is that we want to use intake to expose AzureML datastores, and nothing more. I'm convinced that trying to expose NetCDF or other legacy file types directly to JupyterLab running in AzureML is an antipattern, as is FUSE-mounting to the AzureML filesystem. Instead I think we need to be storing all incoming data in TileDB or Zarr (that is, cloud-optimsed data engines), where these TileDB arrays or Zarrs are provided in Azure Storage Containers that are themselves exposed as a datastore for AzureML.

What we need an intake catalogue to do, then, is to catalogue all the datastores available at the top level, and then also catalogue all the datasets provided per datastore, as well as encapsulating all the boilerplate code needed to connect to a datastore.

That is:
```python
subscription_id = "<sub_id>"
resource_group = "<rg_name>"
workspace_name = "<ws_name>"
datastore_name = "<datastore>"

workspace = Workspace(subscription_id, resource_group, workspace_name)
datastore = Datastore.get(workspace, datastore_name)

uri = f'azure://{datastore.container_name}'

cfg = tiledb.Config()
cfg['vfs.azure.storage_account_name'] = datastore.account_name
cfg['vfs.azure.storage_account_key'] = datastore.account_key
cfg["vfs.s3.use_multipart_upload"] = 'false'
ctx = tiledb.Ctx(config=cfg)

tiledb_name = f"{<tiledb_path>}/{<tiledb_array_name>}"

reader = nctotdb.TileDBReader(tiledb_name, container=container_name,
                              data_array_name=data_array_name, ctx=ctx)
```

all gets encapsulated in the intake catalogue instead:

```python
reader = intake.cat.tiledb.read()
```

This is what we need.

**Comment by tam203** [comment]
My two cent.

* I would avoid AzureML datastores unless we see a need that can't be fulfilled with a more cloud-agnostic approach.
* I agree with @DPeterK about mounting *BUT* I think it's important that we don't let perfect be the enemy of the good (or even just 'the something'). We have to get this done in Oct to give us Nov to test out and bug fix. That means we need things that work now not research projects. This is just the first step, we can iterate and improve as we go. I think if it's working then the ZarrHypothetic might be a good first stab. I don't know where we are with TileDB but I don't feel we 'solved' it last time we took at a run at it and I don't feel we have the time to do so now (but might after AGU, also I still hope we can get TileDB to do most of this for us with some help from HQ but that's further down the line.)

**Comment by DPeterK** [comment]
> I would avoid AzureML datastores unless we see a need that can't be fulfilled with a more cloud-agnostic approach.

I completely disagree. We're deliberately making something for/with Microsoft tools that will only work with AzureML; why now say we need to be cloud-agnostic?? We're spawning AzureML resource, so let's make full use of AzureML, including datastores that are really helpful for encapsulating private data like account keys. Why are we letting perfect be the enemy of the good here when it's not allowed to be in the second bullet point??

> I agree about mounting BUT

We literally cannot mount NetCDF files into AzureML. It's a non-starter. I've proposed something that's working here - we can easily access TileDB or Zarr from AzureML, so why wouldn't we? It's something and it's good, and it's working now.
